packages/zksend/src/experimental/wallet/index.ts

```diff
@@ -337,9 +347,21 @@ export function registerStashedWallet(
 		// Ignore errors
 	}
 
+	let metadata = {};
+	if (useDynamicMetadata) {
+		try {
+			const response = await fetch(METADATA_API_URL);
+			if (!response.ok) throw new Error('Failed to fetch wallet metadata');
+			metadata = await response.json();
```

Comment:
To control the visibility of this, I think one of the pieces of metadata should be `enabled: boolean` so that the response can dictate if the wallet registers or not.